Enhance application configuration and dependencies

ChrisChae00 opened the pull request:
- Added `slack_webhook_url` to `AppState` for optional Slack integration.
- Updated `Cargo.toml` to include `reqwest` with JSON support for HTTP requests.
- Introduced a new `slack` module in the services for future Slack-related functionalities.

ChrisChae00 committed:
Enhance application configuration and dependencies

- Added `slack_webhook_url` to `AppState` for optional Slack integration.
- Updated `Cargo.toml` to include `reqwest` with JSON support for HTTP requests.
- Introduced a new `slack` module in the services for future Slack-related functionalities.

ChrisChae00 committed:
formatting

tonypierre commented on src/config.rs:
```diff
@@ -35,6 +36,7 @@ impl AppState {
             _ => env::var("MONGO_URI").expect("MongoURI not set"),
         };
         let jwt_secret = env::var("JWT_SECRET").expect("JWT_SECRET not set");
+        let slack_webhook_url = env::var("SLACK_WEBHOOK_URL").ok();
```
SHould be under company or team

tonypierre commented on src/services/slack.rs:
```diff
@@ -0,0 +1,99 @@
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct SlackNotification {
```
Let's move this to the model folder

tonypierre commented on src/services/slack.rs:
```diff
@@ -0,0 +1,99 @@
+    pub priority: Option<String>,
+}
+
+pub async fn send_assignee_notification(
```
MAke these traits

ChrisChae00 committed:
Feature/slack user model (#48)

* Add Slack user ID management

- Introduced `slack_user_id` field in the `User` model to store Slack user identifiers.
- Added `UpdateSlackIdPayload` struct for handling updates to the Slack user ID.
- Implemented `handle_update_slack_id` function to process PATCH requests for updating the Slack user ID.
- Updated routing to include a new endpoint for updating Slack user IDs.
- Enhanced the `update_user_slack_id` service function to handle database updates for the new field.

* fomatting

* Feature/slack notification trigger (#49)

* Refactor edit_card function to use AppState and add Slack notification on assignee change

- Updated the `edit_card` function to accept `AppState` instead of `Client` for better state management.
- Implemented logic to send a Slack notification if the card's assignee changes during the edit process.
- Introduced a new `get_user_by_username` function to retrieve user details from the database.

* fix: Resolve compilation errors and improve error handling

- Fix Axum route path syntax (:user_id → {user_id})
- Fix test import paths for proper compilation
- Make create_slack_message function public for testing
- Improve error messages with more specific details
- Update AppState initialization in tests

* formatting and clippy check

* Refactor Slack notification handling (#50)

* Refactor add_card function to use AppState and enhance Slack notification handling

- Updated the `add_card` function to accept `AppState` instead of `Client` for improved state management.
- Implemented Slack notification logic when an assignee is set, including error handling for notification failures.
- Enhanced the Slack message format to provide clearer information about the assigned card.

* Enhance user details response in handle_get_me function

- Added `slack_user_id` to the JSON response in the `handle_get_me` function to include the Slack user identifier for the authenticated user.

* Chore/slack review fixes (#51)

* Refactor Slack notification handling and enhance team model

- Removed `slack_webhook_url` from `AppState` and added it to the `Team` model for better organization of team-specific settings.
- Updated the `add_card` and `edit_card` functions to utilize the team's `slack_webhook_url` for sending notifications, improving the logic for when notifications are sent based on team membership.
- Introduced a new `SlackNotifier` trait and refactored the Slack notification logic to streamline message creation and sending.

* Enhance team retrieval and Slack webhook URL validation

- Updated the `handle_get_team` function to conditionally hide the Slack webhook URL for non-leader users.
- Improved the `update_team` function to validate the Slack webhook URL, ensuring it is not empty and starts with the correct prefix, enhancing error handling for invalid URLs.

* remove debugging logs and check formatting

ChrisChae00 committed:
Refactor Slack notification handling in card services

- Replaced the `SlackWebhookNotifier` with a direct call to `send_assignee_notification` in both `add_card` and `edit_card` functions for improved clarity and efficiency.
- Updated the `send_assignee_notification` function to accept `SlackNotificationPayload` instead of a custom `SlackNotification` type, streamlining the message creation process.
- Removed the `SlackNotifier` trait and associated methods to simplify the notification logic.

ChrisChae00 committed:
Enhance Slack message formatting and update notification structure

- Made the `create_slack_message` function public to facilitate testing.
- Improved the Slack message format by adding a divider and separating the user assignment message from the card title.
- Updated tests to reflect changes in the message structure and ensure correct block counts in the generated messages.
- Refactored notification tests to use `SlackNotificationPayload` for consistency.

ChrisChae00 committed:
Refactor Slack notification handling and introduce SlackNotifier trait

- Replaced direct calls to `send_assignee_notification` with the new `SlackNotifier` trait in `add_card` and `edit_card` functions for improved modularity.
- Implemented `SlackWebhookNotifier` struct to encapsulate Slack notification logic, enhancing code organization and reusability.
- Updated tests to reflect changes in the notification structure and ensure proper functionality.

ChrisChae00 committed:
fix: minor test error

ChrisChae00 committed:
remove: unused method

ChrisChae00 committed:
format
